botocore/signers.py

```diff
@@ -196,6 +208,65 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         return request.url
 
 
+class CloudFrontSigner(object):
+
+    @staticmethod
+    def b64encode(content):
+        return base64.b64encode(content).translate(maketrans(b'+=/', b'-_~'))
+
+    def __init__(self, key_id, private_key):
+        """Create a CloudFrontSigner.
+
+        :type key_id: str
+        :param key_id: The CloudFront Key Pair ID
+
+        :type private_key: str
+        :param private_key: The private key string used for signing.
+        """
+        self.key_id = key_id
+        self.rsa_pk = rsa.PrivateKey.load_pkcs1(private_key.encode('utf8'))
+
+    def sign(self, url, expires, starts=None, ip_address=None):
+        """Sign a given url.
+
+        :type url: str
+        :param url: The URL of the protected object.
+
+        :type expires: int
+        :param expires: The expiry time of the URL. Format is a timestamp.
+
+        :type starts: int
+        :param starts: The start time of the URL. Format is a timestamp.
+
+        :type ip_address: str
+        :param ip_address: Use 'x.x.x.x' for an IP, or 'x.x.x.x/x' for a subnet
```

Comment:
OK then. Incorporating your suggestions, how about this?

1. Let's change the major method to require a policy as input. This one will be future-proof, in a sense that caller can always custom their policy to adapt to future CloudFront requirement.

        def generate_signed_url(self, url, policy):
            pass

2. And let's also have an OPTIONAL policy builder sitting inside this Signer as a helper. The (current) implementation is merely a dozen lines of code, so separating it into another module or even another repo seems like an overkill.

        def build_policy(self, url, expires, starts=None, ip_address=None):
            pass

Eventually the usage pattern would become:

    signer = CloudFrontSigner(key_id, private_key)

    # The following is optional.
    policy = signer.build_policy(url, expires, starts=foo, ip_address=bar)
    # You can always skip above and provide your own.
    policy = {...}

    signed_url = signer.generate_signed_url(url, policy)

I hope this can give us the best of both worlds. I will start working on this one, and also incorporate other review feedbacks. Thanks!